src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java

```diff
@@ -162,11 +162,12 @@ private static void createStdyDscr(XMLStreamWriter xmlw, DatasetDTO datasetDto)
 
         String persistentAuthority = datasetDto.getAuthority();
         String persistentId = datasetDto.getIdentifier();
+        String persistentSeparator = datasetDto.getSeparator();
 
-        String pid = persistentProtocol + ":" + persistentAuthority + "/" + persistentId;
+        String pid = persistentProtocol + ":" + persistentAuthority + persistentSeparator + persistentId;
         String pidUri = pid;
         //Some tests don't send real PIDs - don't try to get their URL form
-        if(!pidUri.equals("null:null/null")) {
+        if(!pidUri.equals("null:nullnullnull")) {
             pidUri= PidUtil.parseAsGlobalID(persistentProtocol, persistentAuthority, persistentId).asURL();
```

Comment:
Can you avoid line 167 by checking for parseAsGlobalId returning null here and setting pid/pidUri using GlobalId if it exists and to the constant "null:nullnullnull" if it doesn't? (Mostly a minor style thing but it avoids writing out unexpected invalid identifier entries as well.)

Reply:
Done!